deepmd/pt/model/model/model.py

```diff
@@ -11,61 +13,14 @@
 )
 
 
-# trick: torch.nn.Module should not be inherbited here, otherwise,
-# the abstract method will override the method from the atomic model
-# as Python resolves method lookups using the C3 linearisation.
-# See https://stackoverflow.com/a/47117600/9567349
-# Take an example, this is the situation for only inheriting make_model():
-#       torch.nn.Module        BaseAtomicModel        make_model()
-#             |                       |                    |
-#             -------------------------                    |
-#                         |                                |
-#                    DPAtomicModel                      BaseModel
-#                         |                                |
-#                make_model(DPAtomicModel)                 |
-#                         |                                |
-#                         ----------------------------------
-#                                           |
-#                                         DPModel
-#
-# The order is: DPModel -> make_model(DPAtomicModel) -> DPAtomicModel ->
-# torch.nn.Module -> BaseAtomicModel -> BaseModel -> make_model()
-#
-# However, if BaseModel also inherbits from torch.nn.Module:
-#         torch.nn.Module                      make_model()
-#                |                                   |
-#                |---------------------------        |
-#                |                          |        |
-#                |      BaseAtomicModel     |        |
-#                |            |             |        |
-#                |-------------             ----------
-#                |                              |
-#           DPAtomicModel                   BaseModel
-#                |                              |
-#                |                              |
-#       make_model(DPAtomicModel)               |
-#                |                              |
-#                |                              |
-#                --------------------------------
-#                         |
-#                         |
-#                      DPModel
```

Comment:
I think it might be helpful to document the new design pattern somewhere appropriate for reference.

Reply:
Fixed in https://github.com/deepmodeling/deepmd-kit/pull/3438/commits/965f471876d1ba69013eea0c4387ba7502f4056e.